### PR TITLE
Added the bootstrap modal for creating a new room. 

### PR DESCRIPTION
**Pull request by namrataaaa**
Changed the style.css file too and imported jquery and popper required for the data attributes of modal to function properly. Closes issue #118

### CODE REVIEW

**Comment by LalitNM** [comment]
I am listing few changes required.

- You need to improve the UX of the modal(you can see #76, this pull request was also for adding modal)
A screenshot from #76.
image


- And when a room already exists your modal just collapses, it should show a message like shown in the screenshot below.

image

- In addition to this I think there are `.fadeIn` and `.fadeOut` in `client.js`. I don't think they are needed anymore.

**Comment by LalitNM** [comment]
@namrataaaa I resolved conflicts, now you are good to go.

**Comment by namrataaaa** [comment]
> @namrataaaa I resolved conflicts, now you are good to go.

Thank you. I have started working on the changes you requested.

**Comment by namrataaaa** [comment]
Hi, @LalitNM I have made the changes you requested. Improved the UX and the modal is showing the error message when a room already exists. However, after a room is being added, I am unable to dismiss the modal. I tried using ```modal('hide')``` function in client.js file but it did not work. Also, I have observed that the comparison of room names was case-sensitive. I tried to change it to uppercase and then compare but it also didn't work. I would also like to show a small notification to the user when a room is successfully created. Please help me on how I can achieve these changes.

File: app.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ io.on('connection', function(socket) {
 		for(var i = 0; i<num_rooms; ++i) {
 
 			//if room name is taken
-			if(rooms[i].name == data.room_name) {
+			if(rooms[i].name.val().toUpperCase() === data.room_name.val().toUpperCase()) {
```
Comment (LalitNM) [inline app.js]: Any specific reason to change this? It is creating bug that modal doesn't collapses.

File: package.json [hunk 2]
```diff
@@ -20,8 +20,8 @@
...
   },
   "dependencies": {
+    "ejs": "^2.5.5",
     "express": "^4.14.0",
-    "socket.io": "^1.7.2",
-    "ejs": "^2.5.5"
+    "socket.io": "^2.3.0"
   }
-}
+}
```
Comment (LalitNM) [inline package.json]: Do not update these dependencies, because at present heroku doesn't support latest versions, app crashes when we update dependencies. You can remove this file.

File: public/script.js [hunk 3]
```diff
@@ -40,16 +40,13 @@ function search() {
   });
 };
 
-function newRoom() {
-  $("#room").css("display", "block");
-};
 
-$("#cancel").click(function() {
+$("#cancel").on('click',function() {
```
Comment (LalitNM) [inline public/script.js]: As far as I can see, you removed `#cancel` and `#sumbit`, you can remove this.

File: public/script.js [hunk 4]
```diff
@@ -40,16 +40,13 @@ function search() {
...
   $(this).parent().parent().hide();
 });
 
 function submitRoom() {
-  createRoom(); // used in client.js to pass details to app.js
+  createRoom(); // used in client.js to pass details to app.js
```
Comment (LalitNM) [inline public/script.js]: Please try not to make some unnecessary changes.

File: public/style.css [hunk 5]
```diff
@@ -402,46 +377,17 @@ body {
...
-}
+} */
 
-#roomName {
+/* #roomName {
   border: none;
-  border-bottom: 1.5px solid #000;
-}
+  border-bottom: 1px solid #000;
+} */
```
Comment (LalitNM) [inline public/style.css]: Can you remove this unnecessary commented code?

File: .gitignore [hunk 6]
```diff
@@ -1 +1,119 @@
...
+
+# yarn v2
+.yarn/cache
+.yarn/unplugged
+.yarn/build-state.yml
+.yarn/install-state.gz
+.pnp.*
+
+# For macOS
+.DS_Store
```
Comment (LalitNM) [inline .gitignore:119]: This all is already there in master branch, you need to rebase your branch with our master branch(see instructions in contributing.md)

File: views/head_imports.ejs [hunk 7]
```diff
@@ -3,4 +3,8 @@
 <link rel="stylesheet" href="/style.css">
 <link rel="stylesheet" href="/emojify.css">
 <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.4.5/socket.io.js"></script>
-<script src="/client.js"></script>
\ No newline at end of file
+<script src="/client.js"></script>
+
+<script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
+<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
```
Comment (LalitNM) [inline views/head_imports.ejs:10]: As you can see we are using cloudflare for cdns, please stick to this.

File: views/index.ejs [hunk 8]
```diff
@@ -1,96 +1,123 @@
...
+          <input type="text" id="searchId" placeholder="Search..." class="searchtext form-control" />
+          <hr class="mb-0" />
+        </div>
+        <!-- Overflow style div -->
+        <div class="room-list">
+          <div class="card-columns m-2" name="people">
+            <div class="card active person" data-chat="lobby" id="lobby" onclick="showRoom(this)">
+              <div class="card-body">
+                <h5 class="card-title name">Lobby</h5>
+                <p class="card-text preview">Central Lobby</p>
```
Comment (LalitNM) [inline views/index.ejs:29]: This all is also unnecessary changes, please use `save without formatting in your editor or change formatting rules of your editor.

**Review by LalitNM — CHANGES_REQUESTED** [review]
I tried to answer all your questions in my review, only one regarding notification left, for this, do not add this functionality now, I will create a separate issue for this when somebody will fix #121.

**Comment by namrataaaa** [comment]
Hi, @LalitNM I made all the changes you requested and also rebased my branch on master. And I was finally able to dismiss the modal once a room was created. Thank you for suggesting those changes.

**Review by LalitNM — APPROVED** [review]

**Comment by LalitNM** [comment]
@namrataaaa by mistake I deleted your files instead of editing them, can you run `git push -f origin feature-modal` on your system. This will restore the previous version of this pull request.

**Comment by LalitNM** [comment]
Merged :rocket: @namrataaaa. Good work on this.

**Comment by namrataaaa** [comment]
> Merged 🚀 @namrataaaa. Good work on this.

Thanks, @LalitNM. I got to learn a lot and had a great experience. Can you please add the label ```hacktoberfest-accepted``` to this PR?

**Comment by LalitNM** [comment]
I have already added `hacktoberfest` topic to whole repository. Still if you are saying, then here you go...
